lib/migration.pm

```diff
@@ -110,5 +111,34 @@ sub record_disk_info {
     assert_script_run 'df -h > /tmp/df.txt';
     upload_logs '/tmp/df.txt';
 }
+
+# System check after snapper rollback
+sub check_rollback_system {
+    # Check if repos are rolled back to correct version
+    script_run("zypper lr -u | tee /dev/$serialdev");
+    my $incorrect_repos = script_output("
+        version=\$(grep VERSION= /etc/os-release | cut -d'=' -f2 | cut -d' ' -f1 | sed 's/\"//g')
+        base_version=\$(echo \$version | cut -d'-' -f1)
+        zypper lr | cut -d'|' -f3 | gawk '/SLE/ || /openSUSE/' | sed \"/\$version\\|Module.*\$base_version/d\"
+    ", 100);
+    record_info('Incorrect Repos', $incorrect_repos, result => 'fail') if $incorrect_repos;
+
+    # Check SUSEConnect status for SLE
+    if (is_sle) {
```

Comment:
how about an early return from this `sub run`? IMHO lower indendation (== complexity) should be preferred, e.g.:

```perl
return unless is_sle;
…
```

but it's your choice :)